lib/telephony/pinpoint/sms_sender.rb

```diff
@@ -105,14 +107,15 @@ def phone_info(phone_number)
       def client_configs
         @client_configs ||= Telephony.config.pinpoint.sms_configs.map do |sms_config|
           credentials = AwsCredentialBuilder.new(sms_config).call
```

Comment:
In retrospect, I don't think I should have made this be a `configs.map` with a `AwsCredentialBuilder.call` (eagerly allocating credentials)

A few other options:

- `client_configs` becomes `each_client_config(&block)` and does some like, try/catch stuff (messy because it might involve interleaving retry-skipping logic with the block?)
- Update the `ClientConfig` struct to make `.client` lazily allocate the client

Reply:
Made an attempt at lazy allocation of credentials: https://github.com/18F/identity-telephony/pull/42, LMK if that looks good and I can try to help patch that back to this PR

Reply:
Definitely, that's a great idea. Also saves an unnecessary call to STS since we'd do one for each config even if we end up using the first.

Reply:
Yup the extra call to STS was the main motivation.